File: CSharp/DP.Domain/Samples/Interpreter/StoreExpression.cs
namespace DP.Domain.Samples.Interpreter
{
    /// <summary>
    /// NonterminalExpression
    /// </summary>
    public class StoreExpression : IExpression
    {
        public void Interpret(Context context)
        {
            context.Output.Store.Id = context.Input.Substring(0, 4).Trim();
            context.Output.Store.Name = context.Input.Substring(4, 20).Trim();
        }
    }
}

File: Python/Samples/Interpreter/StoreExpression.py
from datetime import datetime
from decimal import Decimal 
from Models import PayData, Vip, Store, Context
from Expression import Expression

class StoreExpression(Expression):
     def interpret(self,context=Context):
         context.output.store.id = context.input[0:4].strip()
         context.output.store.name = context.input[4:24].strip()